### PR TITLE
(DOCSP-12508): Update extlinks to match Snooty definitions

### PR DESCRIPTION
**Pull request by nlarew**
This is a sibling PR that preps the Realm docs to match the updated Snooty config in https://github.com/mongodb/snooty-parser/pull/226

## Pull Request Info

Some definitions for extlink roles in snooty ([rstspec.toml](https://github.com/mongodb/snooty-parser/blob/master/snooty/rstspec.toml) don't quite match our legacy definitions. Rather than renaming these in the centralized repo, this PR updates our local `conf.py` and all references to match.

### Issue JIRA link:

- (DOCSP-12508): Centralize extlink definitions

### Docs staging link (requires sign-in on MongoDB Corp SSO):

(probably need to use the diff view for this PR but staged if you want to sanity check)

- [Realm Docs](https://docs-mongodbcom-staging.corp.mongodb.com/realm/nlarew/snooty/index.html)

### CODE REVIEW

File: conf.py [hunk 1]
```diff
@@ -99,7 +99,7 @@
     'admin-api-endpoint': ('https://docs.mongodb.com/realm/admin/api/v3/#%s', ''), # sphinx_openapi doesn't support endpoint refs, so we have to use an extlink instead
     'manual': ('http://docs.mongodb.org/manual%s', ''),
     'atlas': ('https://docs.atlas.mongodb.com%s',''),
-    'datalake': ('https://docs.mongodb.com/datalake/%s',''),
+    'adl': ('https://docs.mongodb.com/datalake/%s',''),
```
Comment (ccho-mongodb) [inline conf.py:102]: I think you want to omit the slash if you're looking to match this:
https://github.com/mongodb/snooty-parser/blob/master/snooty/rstspec.toml#L1070

File: source/mongodb/link-a-data-source.txt [hunk 2]
```diff
@@ -62,7 +62,7 @@ For the most part, you can interact with a linked Data Lake just as
 you would with a linked Atlas cluster. However, there are some
 caveats to keep in mind when working with a Data Lake:
 
-- Data Lakes :datalake:`do not support write operations <supported-unsupported/mql-support>`.
+- Data Lakes :adl:`do not support write operations <supported-unsupported/mql-support>`.
```
Comment (ccho-mongodb) [inline source/mongodb/link-a-data-source.txt:65]: Related to earlier comment, you'll need a leading slash.

**Review by nathan-contino-mongo — APPROVED** [review]
LGTM, good to have these (slightly more) standardized anyway.

**Review by ccho-mongodb — APPROVED** [review]
LGTM but a couple changes needed if you want to match rstspec.toml.
I've run into situations in which directives shared across several repos have been changed by different docs writers without notification which resulted in broken links, so you might be cautious about sharing existing ones. I've recommended separate override files per repo, but not sure what the status of that request is.

File: conf.py (hunk 1 above)
Comment (nlarew) [reply conf.py:102]: Yes good catch!

**Pull request merged**